numpy/core/src/multiarray/iterators.c

```diff
@@ -590,7 +590,11 @@ iter_subscript(PyArrayIterObject *self, PyObject *ind)
     PyArray_ITER_RESET(self);
 
     if (PyBool_Check(ind)) {
-        if (PyObject_IsTrue(ind)) {
+        int istrue = PyObject_IsTrue(ind);
+        if (istrue == -1) {
+            return NULL;
```

Comment:
Looks correct, although it's bit confusing that there is a `goto fail` earlier in the code.  Might make sense to change it there to NULL or here to goto fail.